src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -693,26 +697,34 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
             bool setDefaultTimesOnTokenCreation,
             int tokenLifetimeInMinutes)
         {
-            bool audienceChecked = false;
+            bool descriptorClaimsAudienceChecked = false;
             bool audienceSet = false;
-            bool issuerChecked = false;
+            bool descriptorClaimsIssuerChecked = false;
             bool issuerSet = false;
-            bool expChecked = false;
+            bool descriptorClaimsExpChecked = false;
             bool expSet = false;
-            bool iatChecked = false;
+            bool descriptorClaimsIatChecked = false;
             bool iatSet = false;
-            bool nbfChecked = false;
+            bool descriptorClaimsNbfChecked = false;
             bool nbfSet = false;
 
             writer.WriteStartObject();
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
             {
+                if (!tokenDescriptor.Audience.IsNullOrEmpty() && !tokenDescriptor.Audiences.Contains(tokenDescriptor.Audience))
```

Comment:
We don't need the Contains check, nothing is broken if users have duplicate Audiences in the JsonWebToken.

Reply:
thanks for catching that, I tried removing all the dupe checks but missed this one